modules/mauth-test-utils/src/main/java/com/mdsol/mauth/test/utils/model/UnsignedRequest.java

```diff
@@ -0,0 +1,57 @@
+package com.mdsol.mauth.test.utils.model;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.nio.charset.StandardCharsets;
+
+public class UnsignedRequest {
+  @JsonProperty("verb")
+  String httpVerb;
+  @JsonProperty("url")
+  String urlString;
+  @JsonProperty("body_filepath")
+  String bodyFilepath;
+  @JsonProperty("body")
+  String body;
+
+  String resourcePath;
+  String queryString;
+  byte[] bodyInBytes;
```

Comment:
As mentioned in https://github.com/mdsol/mauth-jvm-clients/pull/124#discussion_r436411441, I find it difficult to track what different test cases there are.

Can we model the possible test cases using a sealed trait? That way it's immediately clear to future readers what test cases we're expecting.

Reply:
I am not sure what the best way to model these cases. `ProtocolTestSuiteHelper.TEST_SPECIFICATIONS` includes all information of mauth-protocol-test-suite which can be reused for the core signing, authentication or httpclient integration testing.

Reply:
There are at least two types of test case:

* Authentication only
* Full test case with the http request, expected `String to Sign` and signature

Reply:
updated in https://github.com/mdsol/mauth-jvm-clients/pull/133/commits/b2625eb91ca7c7de8687f33f2a0eb074dd4fdd59

Reply:
Tests are easier to follow now. Thanks :)